cli/src/base.ts

```diff
@@ -493,4 +506,43 @@ export abstract class BaseCommand extends Command {
       this.warn(`  [${error.code}] ${error.message} at "${error.path.join('.')}"`);
     }
   }
+
+  async prompt<name extends string>(
+    options: prompts.PromptObject<name>,
+    flagValue?: boolean | string
+  ): Promise<prompts.Answers<name>> {
+    // If there's a flag, use the value of the flag
+    if (flagValue != null) return { [String(options.name)]: flagValue } as prompts.Answers<name>;
+
+    const { flags } = await this.parse(
+      { strict: false, flags: { ...BaseCommand.noInputFlag, force: Flags.boolean() } },
+      this.argv
+    );
+    const { 'no-input': noInput, force } = flags;
+
+    if (force && options.initial != null && typeof options.initial !== 'function') {
+      return { [String(options.name)]: options.initial } as prompts.Answers<name>;
+    }
+
+    let reason = '';
+
+    if (!process.stdout.isTTY && process.env.NODE_ENV !== 'test') {
+      reason = 'you are not running it in a TTY';
+    } else if (noInput) {
+      reason = 'the --no-input flag is being used';
+    }
+
+    if (reason) {
+      if (options.type === 'confirm') {
+        this.error(
+          `The current command required interactivity, but ${reason}. Use --force to execute it without asking for confirmations.`
```

Comment:
What if I don't want to really force (in terms of overriding) but still answer yes to all confirmation questions?

To me --no-input is an special --force and here it wouldn't fail, it would succeed.

Only if in TTY and --no-input I would fail, and the message would recommend using --no-input instead of --force.

WDYT?

Reply:
--yes sounds great!